### PR TITLE
Log with request ID and transaction ID in daemon framework

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixes: #849 

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@prashanthpai  please review

File: glusterd2/daemon/daemon.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func StartAllDaemons() {
 	}
 
 	for _, d := range ds {
-		if err := Start(d, true); err != nil {
+		if err := Start(d, true, gdlog); err != nil {
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterd2/daemon/daemon.go]: done

File: glusterd2/brick/glusterfsd.go [hunk 2]
```diff
@@ -167,15 +167,15 @@ func errorContainsErrno(err error, errno syscall.Errno) bool {
 // These functions are used in vol-create, vol-expand and vol-shrink (TBD)
 
 //StartBrick starts glusterfsd process
-func (b Brickinfo) StartBrick() error {
+func (b Brickinfo) StartBrick(gdlog log.FieldLogger) error {
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterd2/brick/glusterfsd.go]: done

File: glusterd2/main.go [hunk 3]
```diff
@@ -20,15 +20,15 @@ import (
...
+	"github.com/sirupsen/logrus"
 	flag "github.com/spf13/pflag"
 	config "github.com/spf13/viper"
 	"github.com/thejerf/suture"
 	"golang.org/x/sys/unix"
 )
 
 func main() {
-
+	var log = logrus.New()
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterd2/main.go]: done

File: glusterd2/main.go [hunk 4]
```diff
@@ -162,9 +162,9 @@ func main() {
 	}
 }
 
-func initGD2Supervisor() *suture.Supervisor {
+func initGD2Supervisor(glog logrus.FieldLogger) *suture.Supervisor {
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterd2/main.go]: done

File: glusterd2/daemon/daemon.go [hunk 5]
```diff
@@ -185,7 +185,7 @@ func Stop(d Daemon, force bool) error {
 }
 
 // StartAllDaemons starts all previously running daemons when GlusterD restarts
-func StartAllDaemons() {
+func StartAllDaemons(gdlog log.FieldLogger) {
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterd2/daemon/daemon.go]: done

**Comment by Madhu-1** [comment]
retest this please

File: glusterd2/snapshot/snapshot-utils.go [hunk 6]
```diff
@@ -78,7 +78,7 @@ func MountSnapBrickDirectory(vol *volume.Volinfo, brickinfo *brick.Brickinfo) er
 }
 
 //ActivateDeactivateFunc uses to activate and deactivate
-func ActivateDeactivateFunc(snapinfo *Snapinfo, b []brick.Brickinfo, activate bool) error {
+func ActivateDeactivateFunc(snapinfo *Snapinfo, b []brick.Brickinfo, activate bool, glog log.FieldLogger) error {
```
Comment (prashanthpai) [inline glusterd2/snapshot/snapshot-utils.go]: Missed replacing glog here.

File: plugins/bitrot/transactions.go [hunk 7]
```diff
@@ -42,19 +42,19 @@ func IsBitrotAffectedNode() bool {
 }
 
 // ManageBitd manages the bitrot daemon bitd. It stops or stops/starts the daemon based on disable or enable respectively.
-func ManageBitd(bitrotDaemon *Bitd) error {
+func ManageBitd(bitrotDaemon *Bitd, glog log.FieldLogger) error {
```
Comment (prashanthpai) [inline plugins/bitrot/transactions.go]: glog

File: plugins/bitrot/transactions.go [hunk 8]
```diff
@@ -63,7 +63,7 @@ func ManageBitd(bitrotDaemon *Bitd) error {
 }
 
 // ManageScrubd manages the scrubber daemon. It stops or stops/starts the daemon based on disable or enable respectively.
-func ManageScrubd() error {
+func ManageScrubd(glog log.FieldLogger) error {
```
Comment (prashanthpai) [inline plugins/bitrot/transactions.go]: glog

File: plugins/glustershd/actor.go [hunk 9]
```diff
@@ -17,7 +19,7 @@ const (
 
 type shdActor struct{}
 
-func (actor *shdActor) Do(v *volume.Volinfo, key string, value string) error {
+func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, glog log.FieldLogger) error {
```
Comment (prashanthpai) [inline plugins/glustershd/actor.go]: glog

File: glusterd2/snapshot/snapshot-utils.go (hunk 6 above)
Comment (Madhu-1) [reply glusterd2/snapshot/snapshot-utils.go]: done

File: plugins/bitrot/transactions.go (hunk 7 above)
Comment (Madhu-1) [reply plugins/bitrot/transactions.go]: done

File: plugins/glustershd/actor.go (hunk 9 above)
Comment (Madhu-1) [reply plugins/glustershd/actor.go]: done

**Pull request merged**